specs/dynamicdependencies/sample-2.md

```diff
@@ -0,0 +1,85 @@
+# Sample 2 - Fabrikam app using Contoso's Muffins package with smart class helpers
+
+Variant of Sample 1 using RAII classes to manage packageDependencyId, packageDependencyContext and DLL lifetimes.
+
+```MddPinPackageDependency``` will fail if there are no packages registered for the calling user satisfying the PackageDependency.
+
+## Win32
+
+```c++
+#include <MsixDynamicDependency.hpp>
+#include <wil/resource.h>
+
+class MddPackageDependencyContextAndDll
+{
+public:
+    MddPackageDependencyContextAndDll() = default;
+
+    ~MddPackageDependencyContextAndDll()
+    {
+        (void) LOG_IF_FAILED(MddRemovePackageDependency(m_packageDependencyContext));
+        (void) LOG_IF_WIN32_BOOL_FALSE(FreeLibrary(m_module.get()));
+    }
+
+    HRESULT Add(
```

Comment:
`Add` should probably be more transactional.  That is, do the `AddPackageDep` work into three locals, then `move` them into the members of this object. (Being sure to clear any current `m_packageDependencyContext` along the way.)